### PR TITLE
Added Master yi  Q W R & Passive

### PR DESCRIPTION
**Pull request by godofloli**
The W isn't complete since I couldn't manage to get how to break the loop once the owner moves also couldn't reset the animation so I have to either press ctrl + 1 to change the animation or auto attack

### CODE REVIEW

File: Champions/MasterYi/Passive.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        {
+           
+        }
+        void OnAutoAttack(AttackableUnit target, bool isCrit)
+        {
+            
+            ObjAIBase dsTarget = target as ObjAIBase;
+            //1.5% bonus dmg every 4 attacks
+            _doublestrikeStacks += 1;
+            if (_doublestrikeStacks < 4)
```
Comment (Deudly) [inline Champions/MasterYi/Passive.cs]: Instead of this, you should use `switch` to avoid this redundant code

File: Champions/MasterYi/Q.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        private Champion _owningChampion;
+        private Spell _owningSpell;
+
+
+        public void OnActivate(Champion owner)
+        {
+            _owningChampion = owner;
+        }
+
+        /*private void OnProc(AttackableUnit target, bool isCrit)
```
Comment (Deudly) [inline Champions/MasterYi/Q.cs]: Why is this code commented?

File: Champions/MasterYi/R.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        {
+        }
+        public void OnStartCasting(Champion owner, Spell spell, AttackableUnit target)
+        {
+            spell.spellAnimation("SPELL4", owner);
+            ApiFunctionManager.AddBuffHUDVisual("Highlander", 10.0f, 1, owner, 10.0f);
+        }
+        public void OnFinishCasting(Champion owner, Spell spell, AttackableUnit target)
+        {
+            /*Passive: Champion kills and assists reduce the remaining cooldown of Master Yi's basic
```
Comment (Deudly) [inline Champions/MasterYi/R.cs]: No need to write this here

File: Champions/MasterYi/W.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                    float healthPercentage = owner.GetStats().CurrentHealth / owner.GetStats().HealthPoints.Total;
+                    //Example : 50/100= 0.5 
+                    float missingHealthPercentage = 1.0f - healthPercentage;
+                    //Right now the result would be 0.5 missingHP
+                    //next we have to heal yi for 30/50/70/90/110 (+0.3) ap * 1% for every 1% missing hp
+                    float ap = owner.GetStats().AbilityPower.Total * 0.3f;
+                    float hptorecover = new float[] { 30f, 50f, 70f, 90f, 110f }[spell.Level - 1] + ap;
+                    float bonushealth = hptorecover * missingHealthPercentage;
+
+                    float ffs = (hptorecover + bonushealth);
```
Comment (Deudly) [inline Champions/MasterYi/W.cs]: What's `ffs`?

File: Champions/MasterYi/W.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+using LeagueSandbox.GameServer.Logic.GameObjects;
+using LeagueSandbox.GameServer.Logic.API;
+using LeagueSandbox.GameServer.Logic.GameObjects.AttackableUnits;
+using LeagueSandbox.GameServer.Logic.Scripting.CSharp;
+
+namespace Spells
+{
+    public class Meditate : GameScript
+    {
+        /*restoring 30/50/70/90/110 (+0.3) Health per second for 4 seconds. This healing is increased by 1% for every 1% of Master Yi's
```
Comment (Deudly) [inline Champions/MasterYi/W.cs]: No need to write this

File: Champions/MasterYi/Passive.cs (hunk 1 above)
Comment (godofloli) [reply Champions/MasterYi/Passive.cs]: not really used to work with that just copied the code from vayne's W & edited it

File: Champions/MasterYi/W.cs (hunk 5 above)
Comment (godofloli) [reply Champions/MasterYi/W.cs]: forgot to delete it :p

File: Champions/MasterYi/W.cs (hunk 4 above)
Comment (godofloli) [reply Champions/MasterYi/W.cs]: for freedom sake don't mind it xd it's just a variable name

File: Champions/MasterYi/Q.cs (hunk 2 above)
Comment (godofloli) [reply Champions/MasterYi/Q.cs]: while I was working I was testing some stuff & the tings who doesnt work I just comment them so if I need them later on u can just remove them if u wish to

File: Champions/MasterYi/W.cs (hunk 4 above)
Comment (Deudly) [reply Champions/MasterYi/W.cs]: Variable names need to have sense for the next contributor that reads your code, so they have to be recognized. Better change the name

File: Champions/MasterYi/Passive.cs (hunk 1 above)
Comment (Deudly) [reply Champions/MasterYi/Passive.cs]: A `switch() { }` really fits here, try it

File: Champions/MasterYi/W.cs (hunk 4 above)
Comment (godofloli) [reply Champions/MasterYi/W.cs]: gonna do it right now

File: Champions/MasterYi/Passive.cs (hunk 1 above)
Comment (godofloli) [reply Champions/MasterYi/Passive.cs]: well what I mean is that am not sure how it works so am afraid I'll make it stop working by doing something wrong :/

**Pull request merged**